packages/datadog-instrumentations/src/sequelize.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+
+const shimmer = require('../../datadog-shimmer')
+
+addHook({ name: 'sequelize', versions: ['>=4'] }, Sequelize => {
+  const startCh = channel('datadog:sequelize:query:start')
+  const finishCh = channel('datadog:sequelize:query:finish')
+
+  shimmer.wrap(Sequelize.prototype, 'query', query => function (sql) {
```

Comment:
VERY weird syntax, i didn't know you could even do that, maybe it's worth to declare the functions above with names instead of putting everything inline

Reply:
This is a fairly common pattern actually, but usually it would all be arrow functions, for example `query => sql => {}`. The reason for using regular functions originally was to be able to name them, but I'm not sure if that's relevant here.

Reply:
If we use arrow functions to wrap methods, we will override the `this` parameter into the methods, because we are not going to be able to get the original `this` because in the arrow functions, `this` makes reference to the parent function `this`.